### DISCUSSION TITLE
Packages hidden network connections

### DISCUSSION THREAD

**Discussion by Crysp** [invertase/react-native-firebase#7455] [Q&A] [answered]
Hi! Could you tell is there any hidden network connections inside these packages?

- @react-native-firebase/app
- @react-native-firebase/auth
- @react-native-firebase/firestore
- @react-native-firebase/messaging
- @react-native-firebase/storage

Maybe socket connections... I'm asking because when app has been launched in background on iOS it have high battery impact. I remove all network while app in background, but it still consumes battery

**Comment by mikehardy** [comment]
I'm not in control of firebase-ios-sdk so I cannot state anything with authority, and the SDK changes semi-frequently so anything I say could be wrong with the next release

People's definition of "hidden" may be different from person to person, so I also can't really state anything with authority without defining "hidden"

I have no idea what your functionality is so I can't say what it's doing. You may have accidentally coded an infinite loop

You have not provided version information so I can't say whether there was a bug discovered in firebase-ios-sdk or react-native or some other module you are using that has a known issue with burning battery

All I can say in any attempt to address the technical question of "are there persistent network connections" is maybe, there are a few things you may toggle off that are on by default - I have made a specific personal effort such that every thing the firebase-ios-sdk allows as optional-but-default-on has an exposed toggle in the firebase.json file, you may try toggling those https://rnfirebase.io/app/json-config

Other than that, you would need to provide a reproduction that shows the problem reliably https://stackoverflow.com/help/minimal-reproducible-example

**Reply by Crysp** [reply]
All packages are "18.5.0"

"Hidden" I mean something in native code (not JS) that create network requests without explicitly declaring this in the JS code

For example, this code creates socket connection

```
firestore().collection('messages').doc('id').onSnapshot(() => {});
```

Is this code creates socket connection? Here is misunderstanding because I think it could be implemented with network connection and without it

```
auth().onAuthStateChanged(() => {});
```

firebase.json is good example of hidden connections

**Reply by mikehardy** [reply, marked as answer]
Right - I believe I understand, based exactly on your examples though, it is clear that it will be app-specific (is the app creating firestore listeners? auth listeners?). Those aren't "hidden" by my definition but they might be by yours. I do not wish to argue semantics, I don't think it will be productive. I recommend you use a network monitoring tool and see for yourself what your app does. Profiling can do the same, for CPU usage.

The firebase.json toggles are only hidden in that you will have those connections by default. They are well documented, with a JSON way to turn them off. Is that hidden? Or just "on by default" ? Again, how you define things is important.

I think the only way to satisfactorily answer your question is to monitor an app directly with profiling + network tracing

**Reply by Crysp** [reply]
@mikehardy could you advice how to monitor network connections (including socket)? Because Flipper does not work with RNFB

**Reply by mikehardy** [reply]
Off the device, something like little snitch or wire shark
